R/utilities-parameters.R

```diff
@@ -39,6 +39,55 @@ readParametersFromXLS <- function(paramsXLSpath, sheets = NULL) {
   return(.parametersVectorToList(pathsValuesVector, pathsUnitsVector))
 }
 
+#' Write parameter structure to excel that can be loaded in MoBi
+#'
+#' @param parameterStructure A list containing vectors 'paths' with the full paths to the parameters,
+#' 'values' the values of the parameters, and 'units' with the units the values are in.
+#'
+#' @param paramsXLSpath Path to the excel file
+#' @param sheet (Optional) name of the excel sheet
+#' @param extend If TRUE, the existing excel file will be extended with the new
+#'  parameter structure. If FALSE (default), the existing file will be
+#'  overwritten.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' params <- list(paths = c("Container1|Path1", "Container|Second|Third|Path2"), values = c(1, 2), units = c("", "µmol"))
+#'
+#' writeParameterStructureToXLS(params, "test.xlsx")
+#' }
+#'
+writeParameterStructureToXLS <- function(parameterStructure, paramsXLSpath, sheet = NULL, extend = FALSE) {
+  .validateParametersStructure(parameterStructure, "parameterStructure")
+  # Split full parameter paths into container path and parameter name
+  containerPaths <- unlist(lapply(parameterStructure$paths, \(x){
+    .splitParameterPathIntoContainerAndName(x)$containerPath
+  }), use.names = FALSE)
+  parameterNames <- unlist(lapply(parameterStructure$paths, \(x){
+    .splitParameterPathIntoContainerAndName(x)$parameterName
+  }), use.names = FALSE)
+
+  # Create a data frame with the parameter structure
+  output <- data.frame(
+    "Container Path" = containerPaths,
+    "Parameter Name" = parameterNames,
+    Value = parameterStructure$values,
+    Units = parameterStructure$units,
+    check.names = FALSE
+  )
+
+  # Write the results into an excel file.
+  # Wrap the output data frame into a list and name the list if sheet name
+  # has been provided
+  data <- list(output)
+  if (!is.null(sheet)) {
+    names(data) <- sheet
+  }
+  writeExcel(data = data, path = paramsXLSpath)
```

Comment:
Make `writeExcel` internal? I think (and hope) that no one uses it directly.

Reply:
done in a9e347a